### PR TITLE
[REVIEW] replace size()==0 with empty(), is_empty()

### PR DESCRIPTION
**Pull request by karthikeyann**
- [x] replace size()==0 with empty() for std::vector
- [x] replace size()==0 with is_empty() for cudf::column_view
- [x] add is_empty to dictionary_column_view and string_column_view

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by codereport — APPROVED** [review]
:+1:

File: cpp/src/rolling/rolling.cu [hunk 1]
```diff
@@ -1531,41 +1531,39 @@ std::unique_ptr<column> grouped_time_range_rolling_window_impl(
...
-             : time_range_window_ASC(input,
-                                     timestamp_column,
-                                     group_offsets,
-                                     group_labels,
-                                     preceding_window_in_days * mult_factor,
-                                     following_window_in_days * mult_factor,
-                                     min_periods,
-                                     aggr,
-                                     mr);
+    return (group_offsets.empty()) ? time_range_window_ASC(input,
```
Comment (davidwendt) [inline cpp/src/rolling/rolling.cu:1534]:
```suggestion
    return group_offsets.empty() ? time_range_window_ASC(input,
```

File: cpp/src/rolling/rolling.cu [hunk 2]
```diff
@@ -1531,41 +1531,39 @@ std::unique_ptr<column> grouped_time_range_rolling_window_impl(
...
-             : time_range_window_DESC(input,
-                                      timestamp_column,
-                                      group_offsets,
-                                      group_labels,
-                                      preceding_window_in_days * mult_factor,
-                                      following_window_in_days * mult_factor,
-                                      min_periods,
-                                      aggr,
-                                      mr);
+    return (group_offsets.empty()) ? time_range_window_DESC(input,
```
Comment (davidwendt) [inline cpp/src/rolling/rolling.cu:1551]:
```suggestion
    return group_offsets.empty() ? time_range_window_DESC(input,
```

**Review by davidwendt — APPROVED** [review]
Looks good. Just a couple of extra `( )`.

**Pull request merged**